### DISCUSSION TITLE
X-Forwarded-* Header Assistance

### DISCUSSION THREAD

**Discussion by ZacharyACoon** [9p4/jellyfin-plugin-sso#144] [Q&A]
Services are often reverse proxied behind 1 or multiple load balancers that handle TLS termination and such.

https://media.example.com/
v (X-Forwarded-Proto: https,  X-Forwarded-Host: media.example.com)
http://media.services.example.com/
v
jellyfin

**Problem**
SSO-Auth only sees the $host variable, so when it forwards the user to for example, keycloak, the request is 
`ohRSwp0Zdl_NwmevGafEks&code_challenge_method=S256&client_id=media.example.com&scope=openid%20profile&redirect_uri=http%3A%2F%2Fmedia.services.location.example.com%2Fjellyfin%2Fsso%2FOID%2Fredirect%2Fexample.com`

Host:  http://media.services.example.com
But that's not what the user needs to be redirected too...

:) Thanks in advance.

**Comment by ZacharyACoon** [comment]
I am testing to see if I can work around this by manually setting the $host variable sent to jellyfin...

```
proxy_set_header Scheme $scheme;
proxy_set_header Host $host;
```

**Comment by 9p4** [comment]
This isn't a plugin issue. This is a configuration issue. Jellyfin/ASP manages the population of the forwarded host variables. It seems like you have two layers of reverse proxies? Does your second proxy trust incoming X-Forwarded headers? By default (for security reasons) they don't.

**Reply by ZacharyACoon** [reply]
Yes, two layers.  Site/TLS termination,  then ServiceGroup Termination (media)

      proxy_set_header Host $host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_set_header X-Forwarded-Protocol $scheme;
      proxy_set_header X-Forwarded-Host $http_host;

**Reply by 9p4** [reply]
Does your second proxy trust incoming X-Forwarded headers?

**Comment by ZacharyACoon** [comment]
I believe so?

```
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_set_header X-Forwarded-Protocol $scheme;
      proxy_set_header X-Forwarded-Host $http_host;
```

**Reply by 9p4** [reply]
Can you add

```
proxy_set_header Forwarded $proxy_add_forwarded;
```

**Reply by 9p4** [reply]
Also is the [realip](https://nginx.org/en/docs/http/ngx_http_realip_module.html) module enabled?
